apps/chat/src/providers/AuthProvider.jsx

```diff
@@ -141,11 +154,26 @@ const AuthProvider = ({ children }) => {
     const stsCredentials = response.ChimeCredentials;
     updateUserAttributes(response.ChimeUserId);
     AWS.config.region = appConfig.region;
-    AWS.config.credentials = {
-      accessKeyId: stsCredentials.AccessKeyId,
-      secretAccessKey: stsCredentials.SecretAccessKey,
-      sessionToken: stsCredentials.SessionToken
-    };
+    AWS.config.credentials = new AWS.Credentials(
+        stsCredentials.AccessKeyId,
+        stsCredentials.SecretAccessKey,
+        stsCredentials.SessionToken);
+
+
+    let credentialReceiveTime = Date.now();
+    // In template.yaml the credential role for anon is set to expire in 15 mins
+    var FIFTEEN_MINS = 15 * 60 * 1000;
```

Comment:
This can be a const, also suggest using naming something like: CREDENTIAL_ROLE_EXPIRY_DURATION

Since, If the role expiry time changes in template, then we have to update the time + variable name here.